go/cmd/vtctldclient/command/vreplication/workflow/update.go

```diff
@@ -112,6 +112,7 @@ func commandUpdate(cmd *cobra.Command, args []string) error {
 			TabletSelectionPreference: tsp,
 			OnDdl:                     binlogdatapb.OnDDLAction(onddl),
 			Shards:                    baseOptions.Shards,
+			State:                     binlogdatapb.VReplicationWorkflowState(textutil.SimulatedNullInt), // We don't allow changing this in the client command
```

Comment:
If we're not using the cli, is there documentation for when the zero value isn't sufficient for a no-op? Does it make sense to at least add field level comments to the proto?

Reply:
Good idea! I've added comments here: https://github.com/vitessio/vitess/pull/15621/commits/3dd0e1337795101a0c5fbcc5645ce8312edbb82c

Reply:
That's helpful. Is it backwards compatible if we add `optional` to those fields?

Reply:
In go protobufs everything is optional. And unfortunately it does not support the more recent `optional` behavior where it injects code to detect if a value was explicitly provided in the message. That's what the simulated NULL stuff is all about, because there's literally no way to determine if a value was actually provided or not. I'm not terribly happy with it, but I could not find a less bad option today. I'd love to get rid of the simulated NULL whenever a better option appears.

Reply:
Sorry, my understanding is now dated! The go implementation DOES now support truly optional field behavior: https://github.com/protocolbuffers/protobuf/blob/v3.25.0/docs/field_presence.md

It does that by making the specified type, e.g. string, a pointer (*string). I'm looking into whether or not that's backward compatible now.

Reply:
https://github.com/vitessio/vitess/issues/15627 ❤️

Reply:
Added a TODO in the comments as well to link everything together.